docusaurus/docs/migration-guides/update-from-grafana-versions/v9.x-v10.x.md

```diff
@@ -37,3 +37,69 @@ Most code targeting 9.x will continue to work without any issues. An exception i
 When writing plugins that should run on 9.x, continue to use the Vector interfaces. In this case, when targeting versions 10+, you can now use simple arrays rather than wrapper classes.
 
 To make this transition seamless, we employed the Original JavaScript Sin™. That is, we [extended the native Array prototype](https://github.com/grafana/grafana/blob/v10.0.x/packages/grafana-data/src/types/vector.ts) with several Vector methods. We will atone and undo this in v11, when Vector interfaces and classes are removed.
+
+## Update to React Router v6
+
+Starting from Grafana 10, plugins can start using the v6 of `react-router`. Overall, `react-router` v6 aims to simplify route configuration and provide a more flexible and intuitive API for developers.
+
+If your current plugin version needs to maintain compatibility with Grafana v9, then you can continue to use `react-router@v5` in Grafana v10. Both versions are available for plugins. However, **we strongly encourage developers to update their plugins to use the v6 version `react-router` as soon as possible**, as the v5 version is going to be deprecated in Grafana v11 and subsequently removed.
+
+[Official React Router v5 to v6 migration guide](https://reactrouter.com/en/main/upgrading/v5)
+
+### Update using `@grafana/create-plugin`
+
+Follow the steps below to start using `react-router` v6 in your plugin:
+
+#### 1. Update the build related configuration:
+
+Enable using `react-router@v6` by setting the following feature flag in `<project-root>/.cprc.json`:
+
+```json
+{
+  "features": {
+    "useReactRouterV6": true
+  }
+}
+```
+
+Now update the build configuration using the create-plugin tool:
+
+```bash
```

Comment:
Ideally switch this to also include yarn/pnpm - could copy logic from `migration-guides/update-create-plugin-versions`

Reply:
Good call! Yeah, I will turn it into an `.mdx` and use the `<CodeSnippet>`.